pkg/checks/latency.go

```diff
@@ -214,16 +206,26 @@ func (l *Latency) check(ctx context.Context) map[string]LatencyResult {
 	var wg sync.WaitGroup
 	results := map[string]LatencyResult{}
 
-	l.mu.Lock()
-	l.client.Timeout = l.cfg.Timeout * time.Second
-	l.mu.Unlock()
+	// Extracting a copy of the http.Client from the context at the start of the function allows us
+	// to safely modify its Timeout property for the latency check without affecting other operations.
+	// This is crucial because different checks might require different timeout settings based on
+	// their nature and the targets they're accessing.
+	// By making a copy, we ensure that these modifications are isolated to each specific check,
+	// preventing race conditions and ensuring that each part of the application can specify
+	// its own requirements.
+	// If a check doesn't need to adjust client properties, it could directly retrieve and use
+	// the client from the context just before the request to minimize modifications
+	// and use the default configuration.
+	c := *httpclient.FromContext(ctx)
```

Comment:
I can remove this comment if you want, my intention with this was to only include it as explanation for the PR